src/test/java/org/folio/rest/api/InstanceBulkTest.java

```diff
@@ -0,0 +1,400 @@
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    int totalMoons = 10;
+    int expectedMatches = 1;
+    Map<String, JsonObject> moons = manyMoons(totalMoons,
+      InstanceBulkIdsGetField.ID);
+    createManyMoons(moons);
+
+    String query = urlEncode("keyword all \"Moon #3\"");
+    URL getInstanceUrl = instanceBulkUrl("/ids?type=id&query=" + query);
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+    client.get(getInstanceUrl, TENANT_ID, json(getCompleted));
+
+    Response response = getCompleted.get(5, SECONDS);
+    validateMoonsResponse(response, expectedMatches, moons);
+  }
+
+  @Test
+  public void canGetInstanceBulkOfIdWithQueryWildcard()
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    int totalMoons = 20;
+    int expectedMatches = 11;
+    Map<String, JsonObject> moons = manyMoons(totalMoons,
+      InstanceBulkIdsGetField.ID);
+    createManyMoons(moons);
+
+    String query = urlEncode("keyword all \"Moon #1*\"");
+    URL getInstanceUrl = instanceBulkUrl("/ids?type=id&query=" + query);
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+    client.get(getInstanceUrl, TENANT_ID, json(getCompleted));
+
+    Response response = getCompleted.get(5, SECONDS);
+    validateMoonsResponse(response, expectedMatches, moons);
+  }
+
+  @Test
+  public void canGetInstanceBulkOfIdWithLanguageGer()
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    int totalMoons = 10;
+    int expectedMatches = 7;
+    Map<String, JsonObject> moons = manyMoons(totalMoons,
+      InstanceBulkIdsGetField.ID);
+    createManyMoons(moons);
+
+    String query = urlEncode("languages=\"ger\"");
+    URL getInstanceUrl = instanceBulkUrl("/ids?type=id&query=" + query);
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+    client.get(getInstanceUrl, TENANT_ID, json(getCompleted));
+
+    Response response = getCompleted.get(5, SECONDS);
+    validateMoonsResponse(response, expectedMatches, moons);
+  }
+
+  private void validateMoonsResponse(Response response, int expectedMatches,
+      Map<String, JsonObject> moons) {
+    assertThat(response.getStatusCode(), is(HTTP_OK));
+    validateMoonsResponse(response.getJson(), expectedMatches, moons);
+  }
+
+  private void validateMoonsResponse(JsonObject collection,
+      int expectedMatches, Map<String, JsonObject> moons) {
+    JsonArray ids = collection.getJsonArray("ids");
+    assertThat(ids.size(), is(expectedMatches));
+    assertThat(collection.getInteger("totalRecords"), is (expectedMatches));
+
+    for (int i = 0; i < ids.size(); i++) {
+      JsonObject idObject = ids.getJsonObject(i);
+      assertThat(moons.containsKey(idObject.getString("id")), is(true));
+    }
+  }
+
+  private void setInstanceSequence(long sequenceNumber) {
```

Comment:
We can remove this method. It is needed for some tests for HRID logic.

Maybe I'm missing something.